ONSdigital/design-system
# Fix defects found in autosuggest

### What is the context of this PR?
Two defects have been observed on the autosuggest component.

1) Deleting a selected suggestion that ends with a ")" breaks the component - Caused by regex error caused on the `emboldenMatch` function.
2) When `allowMultiple` is enabled, deleting already selected suggestions are retained when new suggestions are selected - Caused by the store array function not containing enough logic.

### How to review 

On the multiple selections example (components/autosuggest...
-  type and select "Spain (including Balearic Islands)", delete the end bracket and confirm that the component does not error.
- Add a couple of selections, delete some of the existing selections and add and select a new query and confirm that the correct selections are displayed.
